packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/InAppPurchasePlugin+TestOnly.h

```diff
@@ -0,0 +1,28 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import <StoreKit/StoreKit.h>
+#import "FIAPRequestHandler.h"
+#import "InAppPurchasePlugin.h"
+
+@interface InAppPurchasePlugin ()
+
+// Holding strong references to FIAPRequestHandlers. Remove the handlers from the set after
+// the request is finished.
+@property(strong, nonatomic, readonly) NSMutableSet *requestHandlers;
+
+// Callback channel to dart used for when a function from the transaction observer is triggered.
+@property(strong, nonatomic) FlutterMethodChannel *transactionObserverCallbackChannel;
+
+// Transaction observer methods
+- (void)handleTransactionsUpdated:(NSArray<SKPaymentTransaction *> *)transactions;
+- (void)handleTransactionsRemoved:(NSArray<SKPaymentTransaction *> *)transactions;
+- (void)handleTransactionRestoreFailed:(NSError *)error;
+- (void)restoreCompletedTransactionsFinished;
+- (BOOL)shouldAddStorePayment:(SKPayment *)payment product:(SKProduct *)product;
+
+// Dependency Injection
+- (FIAPRequestHandler *)getHandler:(SKRequest *)request;
```

Comment:
I think I'm misunderstanding something sorry - so for example
https://github.com/flutter/packages/blob/de4e7987bad76baf6d67f5362061d67480fbb113/packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/InAppPurchasePlugin.m#L114-L115
What will these lines then look like? How do I reference the handlerFactory and ask it to please spit out another handler?
Additionally, the `request` param is only available within `startProductRequestProductIdentifiers:` , so how would I be able to call it inside the `initWithRequest:` initializer?

Reply:
They would look like this:

```objc
  FIAPRequestHandler *handler = self.handlerFactory();
  [self.requestHandlers addObject:handler];
```

You don't want to call block in your init method.  You want to hold on to it so that whenever you want a handler you can call it.

This is going to help: http://fuckingblocksyntax.com/  ;)

Reply:
doneeee